test/i2c_rtc8563_test/i2c_rtc8563_test.cpp

```diff
@@ -0,0 +1,222 @@
+                        false,                           // SCL internal pullup NOT enabled
+                        GPIO_NUM_21,                     // SDA pin
+                        false,                           // SDA internal pullup NOT enabled
+                        400 * 1000)                      // clock frequency - 400kHz
+    {
+    }
+
+    void App::init()
+    {
+        Application::init();
+
+        init_i2c_rtc8563();
+
+        if (rtc8563_initialized)
+        {
+            set_time();
+            set_alarm();
+            rtc8563->clear_alarm_flag();
+        }
+    }
+
+    void App::tick()
+    {
+        Log::warning(TAG, "============ APP TICK TICK  =============");
+        SystemStatistics::instance().dump();
+        Log::info(TAG, "........................................" );
+
+        if (rtc8563_initialized)
+        {
+            get_time();
+            get_alarm();
+
+            if (is_alarm_active())
+            {
+                alarm_active_count += 1;
+                Log::info(TAG, "The Alarm Active Count = {}", alarm_active_count);
+                clear_alarm_active();
+            }
+        }
+    }
+
+    // Initialize the RTC8563
+    void App::init_i2c_rtc8563()
+    {
+        auto device = i2c0_master.create_device<PCF8563>(0x51);   // AXP i2c device address  0x51
+        Log::info(TAG, "Scanning for BM8563 ---- {}", device->is_present() ? "device found" : "device NOT present");
+
+        if (device->is_present())
+        {
+            rtc8563_initialized = true;
+            rtc8563 = std::move(device);
+        }
+
+        Log::info(TAG, "BM8563 initialization --- {}", rtc8563_initialized ? "Succeeded" : "Failed");
+    }
+
+    // Set the time
+    void App::set_time()
+    {
+        Log::info(TAG, "********** Setting Time *********");
+        PCF8563::RtcTime tm;
+        tm.seconds = 0;
+        tm.minutes = 8;
+        tm.hours24 = 13;
+        tm.days = 25;
+        tm.weekdays = PCF8563::DayOfWeek::Tuesday;
+        tm.months = PCF8563::Month::February;
+        tm.years = 2020;
+
+        if (!rtc8563->set_rtc_time(tm))
+        {
+            Log::error(TAG, "Error setting RTC time");
+        }
+    }
+
+    // Get the time
+    void App::get_time()
+    {
+        PCF8563::RtcTime tm;
+
+        if (rtc8563->get_rtc_time(tm))
+        {
+            // Tue 25 Feb 2020 - 1:08:00 PM
+            Log::info(TAG, "Time = {} {} {} {} - {} ",
+                        PCF8563::DayOfWeekStrings[static_cast<int>(tm.weekdays)],
+                        tm.days,
+                        PCF8563::MonthStrings[static_cast<int>(tm.months)],
+                        tm.years,
+                        rtc8563->get_12hr_time_string(tm.hours24, tm.minutes, tm.seconds));
+        }
+        else
+        {
+            Log::error(TAG, "Error reading RTC time");
+        }
+    }
+
+    // Set alarm - alarm will activate 12 minutes past any hour of any day of any month of any weekday.
```

Comment:
This comment doesn't match with the values set below.

Reply:
If you are referring to this comment
```
// Set alarm - alarm will activate 12 minutes past any hour of any day of any month of any weekday.
```
The comment is correct since I only enable the alarm_minute.
```
tm.ena_alrm_minute = true;
tm.ena_alrm_hour = false;
tm.ena_alrm_day = false;
tm.ena_alrm_weekday = false;
```

Reply:
Oh, ok. A bit confusing, with the other values being set, but ok.